### PR TITLE
First unit branch

### PR DESCRIPTION
**Pull request by ginaserakalala**

### CODE REVIEW

File: src/main/java/com.teenthofabud.sports.football/Association.java [hunk 1]
```diff
@@ -95,7 +95,10 @@ public static Association editName(Long id, String name) {
         COLLECTION.put(id, association);
         return association;
     }
-
+    public static void resetCollection(){
```
Comment (anirbandas18) [inline src/main/java/com.teenthofabud.sports.football/Association.java]: reset is part of testing and not core business logic. core business logic should never allow for reset of entire data

File: src/main/java/com.teenthofabud.sports.football/Player.java [hunk 2]
```diff
@@ -112,6 +112,10 @@ public static Player editPhoneNumber(Long id, String phoneNumber) {
         COLLECTION.put(id, player);
         return player;
     }
+    public static void resetPlayerCollection(){
```
Comment (anirbandas18) [inline src/main/java/com.teenthofabud.sports.football/Player.java]: reset is part of testing and not core business logic. core business logic should never allow for reset of entire data

File: src/test/java/com/teenthofabud/sports/football/AssociationTest.java [hunk 3]
```diff
@@ -13,38 +25,42 @@ void testAddAssociation(){
...
+        assertEquals("test3website.com", editedWebsite.getWebsite());
     }
     @Test
     void testValidate(){
         Association invalidAssociation = Association.builder().name("").email("").website("").build();
         Exception exception = assertThrows(IllegalArgumentException.class, invalidAssociation::validate);
         assertEquals("Name cannot be empty", exception.getMessage());
     }
-
+    @AfterEach
```
Comment (anirbandas18) [inline src/test/java/com/teenthofabud/sports/football/AssociationTest.java]: self managed only in testing scope. should not affect the main service

File: src/test/java/com/teenthofabud/sports/football/PlayerTest.java [hunk 4]
```diff
@@ -14,32 +24,36 @@ void testAddPlayer(){
...
     void testEditPhone(){
-        Player editedPhone = Player.editPhoneNumber(1L, "0404059509504");
-        assertEquals("0404059509504", editedPhone.getPhoneNumber());
+        Player addedPlayer = Player.add(player);
+        Player editedPhone = Player.editPhoneNumber(addedPlayer.getId(), "024567772345");
+        assertEquals("024567772345", editedPhone.getPhoneNumber());
+    }
+    @AfterEach
+    void tearDown(){
+        Player.resetPlayerCollection();
```
Comment (anirbandas18) [inline src/test/java/com/teenthofabud/sports/football/PlayerTest.java]: self managed only in testing scope. should not affect the main service

**Comment by ginaserakalala** [comment]
Updated test cases to include private methods within Association and Player classes @anirbandas18

File: src/test/java/com/teenthofabud/sports/football/PlayerTest.java [hunk 5]
```diff
@@ -14,32 +27,45 @@ void testAddPlayer(){
...
+        Player addedPlayer = Player.add(player);
+        Player editedPhone = Player.editPhoneNumber(addedPlayer.getId(), "024567772345");
+        assertEquals("024567772345", editedPhone.getPhoneNumber());
     }
+    @AfterEach
+    void tearDown() throws Exception{
+        resetStaticFields();
+    }
+    private void resetStaticFields() throws Exception{
+        Field collectionField = Player.class.getDeclaredField("COLLECTION");
```
Comment (anirbandas18) [inline src/test/java/com/teenthofabud/sports/football/PlayerTest.java]: Do not use reflection. Use the service methods to create your own logic for reset

File: src/test/java/com/teenthofabud/sports/football/AssociationTest.java [hunk 6]
```diff
@@ -13,38 +28,51 @@ void testAddAssociation(){
...
         Association invalidAssociation = Association.builder().name("").email("").website("").build();
         Exception exception = assertThrows(IllegalArgumentException.class, invalidAssociation::validate);
         assertEquals("Name cannot be empty", exception.getMessage());
     }
+    @AfterEach
+    void tearDown() throws Exception{
+        resetStaticFields();
+    }
+    private void resetStaticFields() throws Exception{
+        Field collectionField = Association.class.getDeclaredField("COLLECTION");
```
Comment (anirbandas18) [inline src/test/java/com/teenthofabud/sports/football/AssociationTest.java]: Do not use reflection. Use the service methods to create your own logic for reset

**Review by anirbandas18 — CHANGES_REQUESTED** [review]

**Review by anirbandas18 — APPROVED** [review]

**Pull request merged**